jobbergate-cli/tests/subapps/job_submissions/test_app.py

```diff
@@ -0,0 +1,127 @@
+import shlex
```

Comment:
I'm missing unit tests in which exit code is asserted to be 1. Doesn't make sense for this application to have them?

Reply:
That's a fair observation, so I'll try to explain why here.

The core functionality of this module is provided in the `tools` module and the `requests` module. These commands are thin wrappers around those tools, and those have been thoroughly tested already. So, if we add unit tests for error conditions here, it doesn't really provide any additional coverage. The only causes for these commands to fail are:

- tools methods fail (already tested)
- make_request fails (already tested)
- typer rejects params due to type (tested by typer)

Testing those error conditions wouldn't really add much more than redundancy.

All that being said, I *could* write unit tests for those without much issue. I'm just dubious about the value those tests would add.

Reply:
hmmmm your point is interesting. If you think so I guess it's ok to neglect it